#!/usr/bin/env python
PACKAGE = "strategy"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

## Game Start or not
gen.add("game_start",   bool_t,   0, "Game Start or not",  False)
gen.add("test",   bool_t,   0, "test used",  False)
gen.add("defence",   bool_t,   0, "defence",  False)
## Select Team between Cyan or Magenta
side_enum = gen.enum([ gen.const("Attack_Yellow", str_t, "Blue",   "Attack Yellow Goal"),
                       gen.const("Attack_Blue",   str_t, "Yellow", "Attack Blue Goal")],
                     "An enum to set Team")
gen.add("our_side", str_t, 0, "A side parameter which is edited via an enum", "Yellow", edit_method=side_enum)

## Attack_shoot angle
gen.add("atk_shoot_ang", double_t,  0, "Condition of shootale ang", 5, 0, 10)
gen.add("atk_shoot_dis", double_t,  0, "Condition of shootale dis", 300, 0, 500)


## Robot Speed
gen.add("minimum_w", double_t,   0, "minimum speed of angular velocity",  0.2, 0, 5)
gen.add("maximum_w", double_t,   0, "Maximum speed of angular velocity",  50, 10, 100)
# gen.add("kp_w", double_t,   0, "Kp value of angular velocity",  0.4, 0, 1)
# gen.add("ki_w", double_t,   0, "Ki value of angular velocity",  0, 0, 1)
# gen.add("kd_w", double_t,   0, "Kd value of angular velocity",  0, 0, 1)

gen.add("minimum_v", double_t,   0, "minimum speed of velocity",  0.2, 0, 5)
gen.add("maximum_v", double_t,   0, "Maximum speed of velocity",  30, 10, 100)
# gen.add("kp_v", double_t,   0, "Kp value of velocity",  0.4, 0, 1)
# gen.add("ki_v", double_t,   0, "Ki value of velocity",  0, 0, 1)
# gen.add("kd_v", double_t,   0, "Kd value of velocity",  0, 0, 1)

## Strategy Configs
gen.add("ballhandle_dis", int_t,   0, "Condition of Handleing ball", 28, 0, 50)
gen.add("ballhandle_ang", int_t,   0, "Condition of Handleing ball", 5, 0, 50)

exit(gen.generate(PACKAGE, "strategy", "Robot"))
